matrix-org/matrix-js-sdk
# 🚫 Threads beta testing branch

- Replace dir parameter for threads API
- Use proper unstable parameter for threads API
- Fix bug where undefined vs null in pagination tokens wasn't correctly handled
- Fix bug where thread list results were sorted incorrectly
- Use spec version for determining thread support
- Allow removing the relationship of an event to a thread
- Only use experimental dir param if necessary
- Disable promise short circuiting for now
- workaround to be able to test for now
- Make threads conform to the same timeline APIs as any other timeline
- Extract thread timeline loading out of thread class
- implement new thread loading
- Ensure initial events have metadata set, necessary for jumping to events
- fix thread roots not being updated correctly
- fix jumping to events by link
- fix newly created threads not loading correctly
- make typescript-strict checker a bit happier



## Checklist

* [ ] Sign-off given on the changes (see [CONTRIBUTING.md](https://github.com/matrix-org/matrix-js-sdk/blob/develop/CONTRIBUTING.md))





---
This PR currently has none of the required changelog labels.

Add one of: `T-Deprecation`, `T-Enhancement`, `T-Defect`, `T-Task` to indicate what type of change this is plus `X-Breaking-Change` if it's a breaking change.
